app/models/hint.rb

```diff
@@ -22,7 +22,10 @@ class Hint < ApplicationRecord
 
   def self.match(searchterm)
     searchprint = fingerprint(searchterm)
-    Hint.find_by(fingerprint: searchprint)
+    hints = Hint.where(fingerprint: searchprint)
+    # Order matching hints by source; throw away nils; return first remaining
+    # Hint. If there are none, this will return nil.
+    sources.map { |source| hints.find_by(source: source) }.compact[0]
```

Comment:
With large numbers of sources, this may get inefficient due to how ActiveRecord compiles the logic in this method. I am 100% _not_ suggesting a change as I don't think we should have lots of sources in production but since I was curious how it worked I did some testing. Basically, it ends up combining the queries and then doing once each due to the map... so for two sources we get:
```
[2] pry(main)> Hint.match('pubmed')
  Hint Load (0.2ms)  SELECT  "hints".* FROM "hints" WHERE "hints"."fingerprint" = ? AND "hints"."source" = ? LIMIT ?  [["fingerprint", "pubmed"], ["source", "custom"], ["LIMIT", 1]]
  Hint Load (0.1ms)  SELECT  "hints".* FROM "hints" WHERE "hints"."fingerprint" = ? AND "hints"."source" = ? LIMIT ?  [["fingerprint", "pubmed"], ["source", "aleph"], ["LIMIT", 1]]
=> #<Hint:0x007f8975d3fb48
 id: 90332,
 title: "PubMed",
 url: "http://libraries.mit.edu/get/pubmed",
 fingerprint: "pubmed",
 created_at: Wed, 09 Aug 2017 14:01:14 UTC +00:00,
 updated_at: Wed, 09 Aug 2017 14:01:14 UTC +00:00,
 source: "custom">
```
We have a combined index on those two fields so I think we are all good, just wanted to share the results of my curiosity.